### PR TITLE
Update docs with info/examples on setting Out-File width via PSDefaultParameterValues

### PR DESCRIPTION
**Pull request by rkeithhill**
# PR Summary

## PR Context

Addresses issue documented in #7651 in this repo and in [15404](https://github.com/PowerShell/PowerShell/issues/15404) in PowerShell repo.

~This is a work-in-progress. DO NOT MERGE YET.  I wanted to get feedback on my "wording" for 7.2 before propagating it to the other versions.~

Select the area of the Table of Contents containing the documents being changed.

**Conceptual content**
- [ ] Overview and Install
- [ ] Learning PowerShell
  - [ ] PowerShell 101
  - [ ] Deep dives
  - [ ] Sample scripts
  - [ ] Remoting
- [ ] Release notes (What's New)
- [ ] Windows PowerShell
  - WMF, ISE, release notes, etc.
- [ ] DSC articles
- [ ] Community resources
- [ ] Gallery articles
- [ ] Scripting and development
  - [ ] Language Spec
  - [ ] Legacy SDK

**Cmdlet reference & about_ topics**
- [x] Preview content
- [x] Version 7.1 content
- [x] Version 7.0 content
- [x] Version 5.1 content

## PR Checklist

- [x] I have read the [contributors guide][contrib] and followed the style and process guidelines
- [x] PR has a meaningful title
- [x] PR is targeted at the _staging_ branch
- [x] All relevant versions updated
- [x] Includes content related to issues and PRs - see [Closing issues using keywords][key].
- [x] This PR is ready to merge and is not **Work in Progress**
  - If the PR is work in progress, please add the prefix `WIP:` or `[WIP]` to the beginning of the
    title and remove the prefix when the PR is ready.

[contrib]: https://docs.microsoft.com/powershell/scripting/community/contributing/overview
[key]: https://help.github.com/en/articles/closing-issues-using-keywords

### CODE REVIEW

File: reference/7.2/Microsoft.PowerShell.Core/About/about_Redirection.md [hunk 1]
```diff
@@ -220,6 +220,52 @@ encoding. If the file has a different encoding, the output might not be
...
+### Width of output when writing to a file
+
+When you are writing to a file using either `Out-File` or the redirection
+operators, PowerShell formats table output to the file based on the width of
+the console it is running within. For instance, when logging table output
+to file using a command like e.g. `Get-ChildItem Env:\Path > path.log` on a
+system where the console width is set to 80, the output in the file is
+truncated to 80 characters:
+
+```text
```
Comment (sdwheeler) [inline reference/7.2/Microsoft.PowerShell.Core/About/about_Redirection.md:232]:
```suggestion
```Output
```

File: reference/7.2/Microsoft.PowerShell.Core/About/about_Redirection.md [hunk 2]
```diff
@@ -220,6 +220,52 @@ encoding. If the file has a different encoding, the output might not be
...
+Name                         Value
+----                         -----
+Path                         C:\Program Files\PowerShell\7-preview;C:\WINDOWS…
+```
+
+Considering that the console width may be set arbitrarily on systems where
+your script is run, you may prefer that PowerShell format table output to
+files based on a width that you specify instead.
+
+The `Out-File` cmdlet provides a `Width` parameter that allows you to set the
```
Comment (sdwheeler) [inline reference/7.2/Microsoft.PowerShell.Core/About/about_Redirection.md:242]:
```suggestion
The `Out-File` cmdlet provides a **Width** parameter that allows you to set the
```

File: reference/7.2/Microsoft.PowerShell.Core/About/about_Redirection.md [hunk 3]
```diff
@@ -220,6 +220,52 @@ encoding. If the file has a different encoding, the output might not be
...
+width you would like for table output. Rather than having to add
```
Comment (sdwheeler) [inline reference/7.2/Microsoft.PowerShell.Core/About/about_Redirection.md:243]: The description is very focused on table output. It makes it sound like the width only affects tables, but the width applies to any output.

Reword this to make it clearer that the width applies to all output and that you are using Format-Table as an example.

File: reference/7.2/Microsoft.PowerShell.Core/About/about_Redirection.md [hunk 4]
```diff
@@ -220,6 +220,52 @@ encoding. If the file has a different encoding, the output might not be
...
+`-Width 2000` everywhere you invoke `Out-File`, you can use the
+`$PSDefaultParameterValues` variable to set this value for all usages of the
+`Out-File` cmdlet in a script. And since the redirection operators (`>` and
+`>>`) are effectively aliases for `Out-File`, setting the `Out-File:Width`
+parameter for the whole script impacts the formatting width for the
```
Comment (sdwheeler) [inline reference/7.2/Microsoft.PowerShell.Core/About/about_Redirection.md:248]: Should we make it clear that adding `$PSDefaultParameterValues` in a script makes the change local to the script, but running it from the command line is a global change?

File: reference/7.2/Microsoft.PowerShell.Utility/Out-File.md [hunk 5]
```diff
@@ -133,6 +133,26 @@ provider `Alias:`. The `Get-Location` cmdlet displays the complete path for `Ali
...
+param()
+$PSDefaultParameterValues['out-file:width'] = 2000
+
+$logFile = [System.IO.Path]::ChangeExtension($PSCommandPath, ".log")
+
+Get-ChildItem Env:\ > $logFile
+
+Get-Service -ErrorAction Ignore | Format-Table -AutoSize | Out-File $logFile -Append
+
+Get-Process | Format-Table Id,SI,Name,Path,MainWindowTitle >> $logFile
```
Comment (sdwheeler) [inline reference/7.2/Microsoft.PowerShell.Utility/Out-File.md:153]: I suggest making this into a function. As written, it may not be clear that this is a script. Also, if someone copy/paste this to the command line then the preference becomes global to that session. Wrapping it in a function is safer for the reader.

File: reference/7.2/Microsoft.PowerShell.Core/About/about_Redirection.md [hunk 6]
```diff
@@ -220,6 +220,52 @@ encoding. If the file has a different encoding, the output might not be
...
 `Out-File` cmdlet with its `Encoding` parameter.
 
+### Width of output when writing to a file
+
+When you are writing to a file using either `Out-File` or the redirection
+operators, PowerShell formats table output to the file based on the width of
+the console it is running within. For instance, when logging table output
+to file using a command like e.g. `Get-ChildItem Env:\Path > path.log` on a
+system where the console width is set to 80, the output in the file is
+truncated to 80 characters:
```
Comment (sdwheeler) [inline reference/7.2/Microsoft.PowerShell.Core/About/about_Redirection.md:230]:
```suggestion
to file using a command like `Get-ChildItem Env:\Path > path.log` on a system 
where the console width is set to 80 columns, the output in the file is
truncated to 80 characters:
```

**Review by sdwheeler — CHANGES_REQUESTED** [review]
@rkeithhill Thanks for the updates. Looks pretty good.  See my comments and suggested edits.

File: reference/7.2/Microsoft.PowerShell.Core/About/about_Redirection.md (hunk 3 above)
Comment (rkeithhill) [reply reference/7.2/Microsoft.PowerShell.Core/About/about_Redirection.md:243]: My experiments indicate it doesn't impact Format-List output e.g.:
```powershell
$obj = [pscustomobject]@{prop1=("*" * 1000); prop2=("-" * 1000)}
$obj | format-list > obj.log
notepad .\obj.log
```
You'll see in notepad that each line is 1009 in length without doing anything to the default width. I see the same thing if I use `Format-Custom` instead of `Format-List`.   And if I send a string to output like this:
```powershell
$obj | format-list > obj.log
notepad .\obj.log
```
I see a line in Notepad that ends at col 1001.  I'll bring this up in the wg-engine discord channel to verify.

File: reference/7.2/Microsoft.PowerShell.Core/About/about_Redirection.md (hunk 4 above)
Comment (rkeithhill) [reply reference/7.2/Microsoft.PowerShell.Core/About/about_Redirection.md:248]: I think you have the common case of "user copies script from help example" covered below.  I'll turn that example into a function as you suggest.  Do you think that will be sufficient?

Comment (sdwheeler) [reply reference/7.2/Microsoft.PowerShell.Core/About/about_Redirection.md:248]: Add a link to the about topic.

```
For more information about `$PSDefaultParameterValues`, see [about_Preference_Variables](/powershell/module/microsoft.powershell.core/about/about_preference_variables#psdefaultparametervalues)
```

File: reference/7.2/Microsoft.PowerShell.Utility/Out-File.md [hunk 7]
```diff
@@ -133,6 +133,34 @@ provider `Alias:`. The `Get-Location` cmdlet displays the complete path for `Ali
...
+        Get-Process | Format-Table Id,SI,Name,Path,MainWindowTitle >> $logFile
+    }
+    finally {
+        $PSDefaultParameterValues.Remove('out-file:width')
+    }
+}
+
+DemoDefaultOutFileWidth
+```
+
```
Comment (sdwheeler) [inline reference/7.2/Microsoft.PowerShell.Utility/Out-File.md:163]: Add a link to the about topic.

```
For more information about `$PSDefaultParameterValues`, see [about_Preference_Variables](/powershell/module/microsoft.powershell.core/about/about_preference_variables#psdefaultparametervalues)
```

**Review by sdwheeler — CHANGES_REQUESTED** [review]

File: reference/7.2/Microsoft.PowerShell.Utility/Out-File.md (hunk 5 above)
Comment (rkeithhill) [reply reference/7.2/Microsoft.PowerShell.Utility/Out-File.md]: Done.

File: reference/7.2/Microsoft.PowerShell.Core/About/about_Redirection.md (hunk 6 above)
Comment (rkeithhill) [reply reference/7.2/Microsoft.PowerShell.Core/About/about_Redirection.md]: Done

File: reference/7.2/Microsoft.PowerShell.Utility/Out-File.md (hunk 7 above)
Comment (rkeithhill) [reply reference/7.2/Microsoft.PowerShell.Utility/Out-File.md:163]: I've added this link but while looking the rest of this help topic, it seems the link can be shortened to `../Microsoft.Powershell.Core/About/about_preference_variables#psdefaultparametervalues`.  if that's not right, let me know.

File: reference/7.2/Microsoft.PowerShell.Core/About/about_Redirection.md (hunk 4 above)
Comment (rkeithhill) [reply reference/7.2/Microsoft.PowerShell.Core/About/about_Redirection.md:248]: I think the link can be simplified to `about_preference_variables.md#psdefaultparametervalues` based on what I see for the next link in this topic:
```
The `>` operator is not to be confused with the
[Greater-than](about_Comparison_Operators.md#-gt--ge--lt-and--le) comparison
```
If that's not the case let me know and I'll change it.

**Comment by rkeithhill** [comment]
@sdwheeler Thanks for all your feedback & suggestions on this PR.  It's much appreciated.

File: reference/7.2/Microsoft.PowerShell.Core/About/about_Redirection.md (hunk 4 above)
Comment (sdwheeler) [reply reference/7.2/Microsoft.PowerShell.Core/About/about_Redirection.md:248]: Yep, you are correct. That will work.

File: reference/7.2/Microsoft.PowerShell.Utility/Out-File.md (hunk 7 above)
Comment (sdwheeler) [reply reference/7.2/Microsoft.PowerShell.Utility/Out-File.md:163]: I see that you fixed the link by adding the file extension. Links are tricky. The target can be a file path, which requires the `.md` extension. Or they can be a URL path which must NOT have the file extension. Also file paths can be filesystem relative. But URL paths cannot. They can be site relative, which is the format I used in the suggestion. But I like the changes you made.

**Review by sdwheeler — COMMENTED** [review]
OK, these changes look good. Now copy the changes to the other versions and update the PR. 

Thanks for you help with this.

File: reference/7.2/Microsoft.PowerShell.Core/About/about_Redirection.md [hunk 8]
```diff
@@ -220,6 +220,55 @@ encoding. If the file has a different encoding, the output might not be
...
+you will need to pipe the output through `Format-Table -AutoSize` before
+outputting to file.
+
+```powershell
+$PSDefaultParameterValues['out-file:width'] = 2000
+Get-Service | Format-Table -AutoSize > services.log
+```
+
+For more information about `$PSDefaultParameterValues`, see
+[about_Preference_Variables](about_preference_variables.md#psdefaultparametervalues)
```
Comment (sdwheeler) [inline reference/7.2/Microsoft.PowerShell.Core/About/about_Redirection.md:270]:
```suggestion
[about_Preference_Variables](about_preference_variables.md#psdefaultparametervalues).
```

**Review by sdwheeler — APPROVED** [review]

**Comment by sdwheeler** [comment]
@rkeithhill Thanks for the assist.
